xbmc/network/Socket.cpp

```diff
@@ -171,6 +138,42 @@ bool CPosixUDPSocket::Bind(bool localOnly, int port, int range)
   return true;
 }
 
+bool CPosixUDPSocket::CheckIPv6(int port, int range)
+{
+  bool ipv6 = false;
+  CAddress testaddr("::");
+
+  SOCKET testSocket = socket(AF_INET6, SOCK_DGRAM, IPPROTO_UDP);
+  if (testSocket == INVALID_SOCKET)
```

Comment:
What's kodi's philosophy on resetting ```errno```?

Reply:
I did a short spot check and couldn't find any code that resets `errno`.